### PR TITLE
Fix ChatConversationalAgentOutputParser cannot handle markdown in the JSON

### PR DESCRIPTION
**Pull request by micahriggan**
The output parser doesn't handle when the AI's action_input contains markdown, or ``` backticks really, as it's matching on the first set of backticks it finds rather than the last.

Tested with `yarn test:single langchain/src/agents/tests/chat_output_parser.test.ts`

### CODE REVIEW

**Comment by micahriggan** [comment]
For the lint, how should I resolve the no-unused-expressions on the tests calling expect?

File: langchain/src/agents/tests/chat_output_parser.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { ChatConversationalAgentOutputParser } from "../chat_convo/outputParser.js";
```
Comment (nfcampos) [inline langchain/src/agents/tests/chat_output_parser.test.ts:1]:
```suggestion
import { test, expect } from "@jest/globals";
import { ChatConversationalAgentOutputParser } from "../chat_convo/outputParser.js";
```

**Comment by nfcampos** [comment]
@micahriggan I've fixed it, it was missing the `()` for doing the function call

**Comment by NJordan72** [comment]
FWIW, I think [StructuredOutputParser](https://github.com/hwchase17/langchainjs/blob/main/langchain/src/output_parsers/structured.ts) has the same problem

**Comment by micahriggan** [comment]
@NJordan72 agreed, just took a look at that file and it has the same issue.

**Comment by micahriggan** [comment]
Anything else I need to do on this one? @nfcampos

**Comment by nfcampos** [comment]
Thanks, merging this now. The same issue in StructuredOutputParser is resolved in another PR

**Pull request merged**